pythonforandroid/recipe.py

```diff
@@ -24,8 +24,16 @@
 # this import is necessary to keep imp.load_source from complaining :)
 if PY2:
     import imp
-    import_recipe = imp.load_source
+    import warnings
+
+    def import_recipe(module, filename):
+        with warnings.catch_warnings():
+            # ignores warnings raised by hierarchical module names
+            # (names containing dots) on Python 2
+            warnings.simplefilter("ignore", RuntimeWarning)
+            return imp.load_source(module, filename)
```

Comment:
why all the warning suppression when we could just transform to an absolute path?

**UNTESTED**, but dumping both the warnings and the odd branched definitions of `import_recipe` like this would IMHO be both shorter and more readable:

```
def import_recipe(module, filename):
    filename = os.path.join(os.path.abspath(os.path.dirname(__file__), "recipes", filename)
    if PY2:
        return imp.load_source(module, filename)
    else:
        import importlib.util
        if hasattr(importlib.util, 'module_from_spec'):
            # Python 3.5
            spec = importlib.util.spec_from_file_location(module, filename)
            mod = importlib.util.module_from_spec(spec)
            spec.loader.exec_module(mod)
            return mod
        else:
            # Python 3.3 and 3.4:
            from importlib.machinery import SourceFileLoader
            return SourceFileLoader(module, filename).load_module()
```
(unless I am missing something that wouldn't make this feasible)

Reply:
sorry, I guess the transformation of the path should be gated by a check of `os.path.isabs` - but I hope I illustrated the general idea well enough

Reply:
Thanks for the feedback! Yes I also think the way it's currently written is strange, but I didn't aim at refactoring it. And yes I would probably have written it the same way as you did.
Anyway I don't see how this would remove the warnings from the `imp.load_source(module, filename)` call :thinking: 
Could you elaborate?

Reply:
Oh it doesn't? Again I haven't tested it, but I thought the warnings were because it's a relative import. (rather than from an absolute path) or do they appear in any case when using `imp.load_source()`?

Reply:
Oops nevermind I don't know why I thought that :laughing: :laughing:

yeah, stick with the warning suppression then. but I would still suggest reformatting it into a single function, it's just a tad more readable like that IMHO

Reply:
I've refactored, I was really boiling to do it, but I didn't to mix concerns to ease reviewer task. But if you ask for it, there you have it :smile: